Foundation/NSString.swift

```diff
@@ -633,12 +633,16 @@ extension NSString {
         let scanner = Scanner(string: _swiftObject)
         // skip initial whitespace if present
         let _ = scanner.scanCharactersFromSet(.whitespaces)
-        // scan a single optional '+' or '-' character, followed by zeroes
-        if scanner.scanString("+") == nil {
-            let _ = scanner.scanString("-")
-        }
+        // scan a single optional '+' or '-' character
+        let hasSignal = scanner.scanString("+") != nil || scanner.scanString("-") != nil
         // scan any following zeroes
```

Comment:
Looking at the Darwin implementation, it seems we check only the first character for `t T y Y` and return `true`. After that is when we do the scan for `+`/`-` and the digits.

Reply:
@otaviolima Can you switch this with the check in the return statement below? (i.e., check that first, then this.)

Reply:
You mean check in the same way @parkera explained it is in Darwin implementation?

Reply:
Yeah.

Reply:
Yup, doing that right now. ^-^

Reply:
Done.